lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -742,7 +801,15 @@ namespace Js
         Assert(this->errorObject == nullptr);
         if (this->errorObject != nullptr)
         {
-            JavascriptExceptionOperators::Throw(errorObject, scriptContext);
+            if (this->promise != nullptr)
```

Comment:
can this codepath hit? there is an assert above which says Assert(this->errorObject == nullptr); ?